modules/build/src/main/scala/scala/build/Build.scala

```diff
@@ -825,9 +826,15 @@ object Build {
             Seq(ScalacOpt("-scalajs"))
           else Nil
 
-        val scalacReleaseV = releaseFlagVersion
-          .map(v => List("-release", v).map(ScalacOpt(_)))
-          .getOrElse(Nil)
+        val scalacReleaseV =
+          // the -release flag is not supported for Scala 2.12.x < 2.12.5
+          if params.scalaVersion.startsWith("2.12") &&
+            params.scalaVersion.maybeScalaPatchVersion.exists(_ < 5)
```

Comment:
No big deal, but version comparison with `coursier.core.Version` could have worked here, it seems, like
```scala
val sv = coursier.core.Version(params.scalaVersion)
if params.scalaVersion.startsWith("2.12.") && sv.compareTo(coursier.core.Version("2.12.15")) < 0
```